pkg/trace/remoteconfighandler/remote_config_handler_test.go

```diff
@@ -0,0 +1,213 @@
+
+	payload := apmsampling.SamplerConfig{
+		AllEnvs: apmsampling.SamplerEnvConfig{
+			ErrorsSamplerTargetTPS: floatPointer(42),
+		},
+	}
+
+	raw, _ := json.Marshal(payload)
+	config := state.APMSamplingConfig{
+		Config: raw,
+	}
+
+	prioritySampler.EXPECT().UpdateTargetTPS(float64(41)).Times(1)
+	errorsSampler.EXPECT().UpdateTargetTPS(float64(42)).Times(1)
+	rareSampler.EXPECT().SetEnabled(true).Times(1)
+
+	h.onUpdate(map[string]state.APMSamplingConfig{"datadog/2/APM_SAMPLING/samplerconfig/config": config})
+
+	ctrl.Finish()
+}
+
+func TestRareSampler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+
+	agentConfig := config.AgentConfig{RemoteSamplingClient: remoteClient, TargetTPS: 41, ErrorTPS: 41, RareSamplerEnabled: true}
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	payload := apmsampling.SamplerConfig{
+		AllEnvs: apmsampling.SamplerEnvConfig{
+			RareSamplerEnabled: boolPointer(false),
+		},
+	}
+
+	raw, _ := json.Marshal(payload)
+	config := state.APMSamplingConfig{
+		Config: raw,
+	}
+
+	prioritySampler.EXPECT().UpdateTargetTPS(float64(41)).Times(1)
+	errorsSampler.EXPECT().UpdateTargetTPS(float64(41)).Times(1)
+	rareSampler.EXPECT().SetEnabled(false).Times(1)
+
+	h.onUpdate(map[string]state.APMSamplingConfig{"datadog/2/APM_SAMPLING/samplerconfig/config": config})
+
+	ctrl.Finish()
+}
+
+func TestEnvPrecedence(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+
+	agentConfig := config.AgentConfig{RemoteSamplingClient: remoteClient, TargetTPS: 41, ErrorTPS: 41, RareSamplerEnabled: true, DefaultEnv: "agent-env"}
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	payload := apmsampling.SamplerConfig{
+		AllEnvs: apmsampling.SamplerEnvConfig{
+			PrioritySamplerTargetTPS: floatPointer(42),
+			ErrorsSamplerTargetTPS:   floatPointer(42),
+			RareSamplerEnabled:       boolPointer(true),
+		},
+		ByEnv: []apmsampling.EnvAndConfig{{
+			Env: "agent-env",
+			Config: apmsampling.SamplerEnvConfig{
+				PrioritySamplerTargetTPS: floatPointer(43),
+				ErrorsSamplerTargetTPS:   floatPointer(43),
+				RareSamplerEnabled:       boolPointer(false),
+			},
+		}},
+	}
+
+	raw, _ := json.Marshal(payload)
+	config := state.APMSamplingConfig{
+		Config: raw,
+	}
+
+	prioritySampler.EXPECT().UpdateTargetTPS(float64(43)).Times(1)
+	errorsSampler.EXPECT().UpdateTargetTPS(float64(43)).Times(1)
+	rareSampler.EXPECT().SetEnabled(false).Times(1)
+
+	h.onUpdate(map[string]state.APMSamplingConfig{"datadog/2/APM_SAMPLING/samplerconfig/config": config})
+
+	ctrl.Finish()
+}
+
+func floatPointer(f float64) *float64 {
```

Comment:
I thought these pointer helper methods were removed?

Reply:
I removed useless usage where I was dereferencing the returned pointer instead of just creating a float, but there are places in the test where I legitimately need to create float pointers